packages/internal/src/generate/watch.ts

```diff
@@ -28,67 +29,190 @@ import {
+  if (!name.endsWith('Page')) {
+    return false
+  }
+
+  // A page should be in the `web/src/pages` directory.
+  if (!isFileInsideFolder(p, getPaths().web.pages)) {
+    return false
+  }
+
+  return true
+}
+
 const rwjsPaths = getPaths()
 
-const watcher = chokidar.watch('**/src/**/*.{ts,js,jsx,tsx}', {
+const action = {
+  add: 'Created',
+  unlink: 'Deleted',
+  unlinkDir: 'Deleted',
+  change: 'Modified',
+}
+
+const apiWatcher = chokidar.watch('src/**/*.{ts,js}', {
   persistent: true,
   ignored: ['node_modules', '.redwood'],
   ignoreInitial: true,
-  cwd: rwjsPaths.base,
+  cwd: rwjsPaths.api.base,
   awaitWriteFinish: true,
 })
 
-const action = {
-  add: 'Created',
-  unlink: 'Deleted',
-  change: 'Modified',
-}
+const dirWatcher = chokidar.watch('src', {
+  cwd: rwjsPaths.web.base,
+  ignoreInitial: true,
+  awaitWriteFinish: true,
+})
+
+apiWatcher.on('all', async (eventName, p) => {
+  if (!['add', 'change', 'unlink'].includes(eventName)) {
+    return
+  }
+
+  eventName = eventName as 'add' | 'change' | 'unlink'
+
+  const absPath = path.join(rwjsPaths.api.base, p)
+
+  if (isDirectoryNamedModuleFile(absPath)) {
+    if (eventName === 'unlink') {
+      try {
+        fs.unlinkSync(mirrorPathForDirectoryNamedModules(absPath, rwjsPaths)[0])
 
-watcher
+        console.log(
+          action[eventName],
+          'Directory named module:',
+          '\x1b[2m',
+          p,
+          '\x1b[0m'
+        )
+      } catch {
+        // Ignoring any errors for the same reasons as for Cell
+      }
+    } else {
+      generateMirrorDirectoryNamedModule(absPath, rwjsPaths)
+
+      console.log(
+        action[eventName],
+        'Directory named module:',
+        '\x1b[2m',
+        p,
+        '\x1b[0m'
+      )
+    }
+  } else if (isGraphQLSchemaFile(absPath)) {
+    await generateGraphQLSchema()
+    await generateTypeDefGraphQLApi()
+    console.log(action[eventName], 'GraphQL Schema:', '\x1b[2m', p, '\x1b[0m')
+  }
+})
+
+dirWatcher
   .on('ready', async () => {
     console.log('Generating TypeScript definitions and GraphQL schemas...')
     const files = await generate()
     console.log(files.length, 'files generated')
   })
   .on('all', async (eventName, p) => {
-    if (!['add', 'change', 'unlink'].includes(eventName)) {
+    if (!['add', 'change', 'unlink', 'unlinkDir'].includes(eventName)) {
       return
     }
-    eventName = eventName as 'add' | 'change' | 'unlink'
 
-    const absPath = path.join(rwjsPaths.base, p)
+    eventName = eventName as 'add' | 'change' | 'unlink' | 'unlinkDir'
 
-    if (absPath.indexOf('Cell') !== -1 && isCellFile(absPath)) {
-      await generateTypeDefGraphQLWeb()
-      if (eventName === 'unlink') {
-        fs.unlinkSync(mirrorPathForCell(absPath, rwjsPaths)[0])
-      } else {
-        generateMirrorCell(absPath, rwjsPaths)
+    const absPath = path.join(rwjsPaths.web.base, p)
+
+    if (eventName === 'unlinkDir') {
+      if (absPath.endsWith('Cell')) {
```

Comment:
Won't this just always fail with `BazingaCell.js` where it actually ends with Cell.js? 

Also why remove the `isCellFile` check here?

Reply:
This is `unlinkDir`, so it's looking at folder names. So for `BazingaCell.js` `absPath` here would probably be `.../web/src/components/BazingaCell`

Can't use `isCellFile` because it's a folder, not a file

Reply:
Gotcha, yeah didn't think of that!